src/Engine/Manager/RenderManager.cpp

```diff
@@ -203,12 +207,46 @@ void RenderManager::UpdateBufferSize() {
 void RenderManager::RenderWorldEntities(World& world, const glm::mat4& viewMatrix, const glm::mat4& projectionMatrix, Video::RenderSurface* renderSurface) {
     // Render from camera.
     renderer->StartRendering(renderSurface);
+    glm::mat4 lightViewMatrix;
+    glm::mat4 lightProjection;
+    float aspectRatio = static_cast<float>(shadowPass->GetShadowWidth()) / shadowPass->GetShadowHeight();
+
+    for (Component::SpotLight* spotLight : spotLights.GetAll()) {
+        if (spotLight->IsKilled() || !spotLight->entity->enabled)
+            continue;
 
+        if (spotLight->shadow) {
+            Entity* lightEntity = spotLight->entity;
+            lightViewMatrix = glm::inverse(lightEntity->GetModelMatrix());
+            lightProjection = glm::perspective(glm::radians(2.f * spotLight->coneAngle), aspectRatio, 0.5f, 50.0f);
+        }
+    }
+   
     // Camera matrices.
     const glm::mat4 viewProjectionMatrix = projectionMatrix * viewMatrix;
-
     const std::vector<Mesh*>& meshComponents = meshes.GetAll();
-
+    //render shadows
+    { PROFILE("Render Shadows meshes");
+    { GPUPROFILE("Render Shadows meshes", Video::Query::Type::TIME_ELAPSED);
+    { GPUPROFILE("Render Shadows meshes", Video::Query::Type::SAMPLES_PASSED);
+        renderer->PrepareShadowRendering(lightViewMatrix, lightProjection, shadowPass->GetShadowID(), shadowPass->GetShadowWidth(), shadowPass->GetShadowHeight(),shadowPass->GetDepthMapFbo());
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') comma should be followed by whitespace [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=vera%2B%2B%3AT009)